python/ray/serve/autoscaling_policy.py

```diff
@@ -88,16 +89,36 @@ def _calculate_desired_num_replicas(
     return desired_num_replicas
 
 
+@dataclass
 @PublicAPI(stability="alpha")
-def replica_queue_length_autoscaling_policy(
-    curr_target_num_replicas: int,
-    current_num_ongoing_requests: List[float],
-    current_handle_queued_queries: float,
-    config: Optional[AutoscalingConfig],
-    capacity_adjusted_min_replicas: int,
-    capacity_adjusted_max_replicas: int,
-    policy_state: Dict[str, Any],
-) -> int:
+class AutoscalingContext:
+    """The context for an autoscaling policy call."""
+
+    # The AutoscalingConfig which the deployment started with.
+    config: AutoscalingConfig
+    # The number of replicas that the deployment is currently trying to scale to.
+    current_target_num_replicas: int = 0
+    # List of number of ongoing requests for each replica.
+    current_num_ongoing_requests: List[int] = field(default_factory=list)
+    # The number of handle queued queries, if there are multiple handles, the max
+    # number of queries at a single handle will be passed.
+    current_handle_queued_queries: float = 0.0
+    # The min_replica of the deployment adjusted by the target capacity.
+    capacity_adjusted_min_replicas: Optional[int] = None
+    # The max_replica of the deployment adjusted by the target capacity.
+    capacity_adjusted_max_replicas: Optional[int] = None
+    # State of the policy to be used during the call
+    policy_state: Dict[str, Any] = field(default_factory=dict)
+    # The timestamp of last scaled time. Will be None If the deployment have not scaled.
+    last_scale_time: Optional[float] = None
```

Comment:
any reason this shouldn't just be tracked as part of `policy_state`?

Reply:
`policy_state` was more of a after thought, will move `last_scale_time` into `policy_state`

Reply:
I think policy-state should be comprised of 2 pieces:

 - Free form portion (KV store) for policy to persist arbitrary state
 - Well-defined meta information (last updated time, bounded history of previous decisions, etc)